src/Microsoft.Data/PrimitiveColumnContainer.cs

```diff
@@ -69,91 +81,180 @@ public PrimitiveColumnContainer(long length = 0)
                 if (Buffers.Count == 0)
                 {
                     Buffers.Add(new DataFrameBuffer<T>());
+                    NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
                 }
                 DataFrameBuffer<T> lastBuffer = Buffers[Buffers.Count - 1];
                 if (lastBuffer.Length == lastBuffer.MaxCapacity)
                 {
                     lastBuffer = new DataFrameBuffer<T>();
                     Buffers.Add(lastBuffer);
+                    NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
                 }
                 int allocatable = (int)Math.Min(length, lastBuffer.MaxCapacity);
                 lastBuffer.EnsureCapacity(allocatable);
+                DataFrameBuffer<byte> lastNullBitMapBuffer = NullBitMapBuffers[NullBitMapBuffers.Count - 1];
+                lastNullBitMapBuffer.EnsureCapacity(allocatable);
                 lastBuffer.Length = allocatable;
+                lastNullBitMapBuffer.Length = allocatable;
                 length -= allocatable;
                 Length += lastBuffer.Length;
             }
         }
 
-        public void Append(T value)
+        public void Append(T? value)
         {
             if (Buffers.Count == 0)
             {
                 Buffers.Add(new DataFrameBuffer<T>());
+                NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
             }
             DataFrameBuffer<T> lastBuffer = Buffers[Buffers.Count - 1];
             if (lastBuffer.Length == lastBuffer.MaxCapacity)
             {
                 lastBuffer = new DataFrameBuffer<T>();
+                Buffers.Add(lastBuffer);
+                NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
             }
-            lastBuffer.Append(value);
+            lastBuffer.Append(value ?? default);
+            SetBit(Length, value.HasValue ? true : false);
             Length++;
         }
 
+        public bool IsValid(long index) => NullCount == 0 || GetBit(index);
+
+        private void SetBit(long index, bool value)
```

Comment:
Is there a better name than `SetBit` for this method? I don't think it is very descriptive of what it does.  maybe `SetNullIndex`? or something linking it to `Null` support.

Reply:
How about `SetValidityBit`? Unfortunately in Arrow, a `1` implies a valid value and a `0` implies a null value.